### PR TITLE
[bugfix] remove nan loss in manual optimization

### LINKED ISSUE #5123: In manual optimization, loss=NaN is displayed in progress_bar
### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR fixes progress bar displaying NaN loss in manual_optimization.

Fixes #5123

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]
lgtm

**Review by s-rog — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1392,12 +1392,15 @@ def get_progress_bar_dict(self):
...
-        tqdm_dict = {"loss": "{:.3g}".format(avg_training_loss)}
+        if self.trainer.train_loop.automatic_optimization:
+            avg_training_loss = (
+                running_train_loss.cpu().item()
+                if running_train_loss is not None
+                else float("NaN")
+            )
+            tqdm_dict = {"loss": "{:.3g}".format(avg_training_loss)}
+        else:
+            tqdm_dict = {}
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1403]: We can reorder this code to support manual optimization users who set `running_losses` themselves:

```suggestion
        avg_training_loss = None
        if running_train_loss is not None:
            avg_training_loss = running_train_loss.cpu().item()
        elif self.trainer.train_loop.automatic_optimization:
            avg_training_loss = float('NaN)

        tqdm_dict = {}
        if avg_training_loss is not None:
            tqdm_dict["loss"] = f"{avg_training_loss:.3g}"
```

Comment (tchaton) [reply pytorch_lightning/core/lightning.py:1403]: Hey @carmocca, I don't except people using `running_losses` as it is an internal object. But I like the idea :)

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 2]
```diff
@@ -1392,12 +1392,15 @@ def get_progress_bar_dict(self):
...
-            running_train_loss.cpu().item()
-            if running_train_loss is not None
-            else float("NaN")
-        )
-        tqdm_dict = {"loss": "{:.3g}".format(avg_training_loss)}
+        avg_training_loss = None
+        if running_train_loss is not None:
+            avg_training_loss = running_train_loss.cpu().item()
+        elif self.trainer.train_loop.automatic_optimization:
+            avg_training_loss = float('NaN)
```
Comment (rohitgr7) [inline pytorch_lightning/core/lightning.py:1399]:
```suggestion
            avg_training_loss = float('NaN')
```
